lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,273 @@
+            "aggs": {
+                "controllers": {
+                    "terms": {"field": "run.controller", "order": [{"runs": "desc"}]},
+                    "aggs": {"runs": {"max": {"field": "run.start"}}},
+                }
+            },
+        }
+
+        # TODO: the big assumption here involves the index version we're
+        # referencing. While it's possible the root index name could change
+        # at some point, advancing the version is going to happen more
+        # often.
+        #
+        # So ... do we need to change the code here each time? The old
+        # dashboard config is hardcoded for each version, so one option is
+        # to add a version for each index to the pbench-server.cfg file.
+        #
+        # When we have a full live Python server, persisting the template
+        # information in Redis or postgreSQL probably makes sense; maybe
+        # we just hardcode until then?
+        uri_fragment = QueryControllers._genMonthUriRange(
+            prefix, ".v6.run-data.", start, end
+        )
+
+        uri = f"{self.app.config['ES_URL']}/{uri_fragment}/_search"
+        logger.info("POSTING query to {}: {}", uri, payload)
+        try:
+            # query Elasticsearch
+            es_response = requests.post(
+                uri,
+                params={"ignore_unavailable": "true"},
+                headers={
+                    "Content-Type": "application/json",
+                    "Accept": "application/json",
+                },
+                json=payload,
+            )
+        except requests.exceptions.HTTPError as e:
+            logger.exception("HTTP error {} from Elasticsearch post request", e)
+            abort(500, message=f"HTTP error {e} from Elasticsearch")
+        except requests.exceptions.ConnectionError:
+            logger.exception("Connection refused during the Elasticsearch post request")
+            abort(502, message="Network problem, could not post to Elasticsearch")
+        except requests.exceptions.Timeout:
+            logger.exception(
+                "Connection timed out during the Elasticsearch post request"
+            )
+            abort(
+                504, message="Connection timed out, could not post to Elasticsearch",
+            )
+        except requests.exceptions.InvalidURL:
+            logger.exception(
+                "Invalid url {} during the Elasticsearch post request", uri
+            )
+            abort(
+                500,
+                message=f"Invalid Elasticsearch url {uri}, could not complete the post request",
+            )
+        except Exception as e:
+            logger.exception(
+                "Exception {!r} occurred during the Elasticsearch post request", e
+            )
+            abort(500, message=f"Could not post to Elasticsearch: {e!r}")
+        else:
+            es_json = es_response.json()
+            logger.info("{} hits found", len(es_json["hits"]["hits"]))
```

Comment:
We might want to add some context value to match the POST log early with this log message.  When we are searching through lots of logs on the server, it will help to match them up.

Reply:
I had thought at one point that an info log for each call might be useful; we can get the API log from httpd, but that doesn't have any payload data if we really want to record that. The tradeoff is that we can get a lot of API hits. I've always gone back and forth on that tradeoff; this time I decided not to. But then I went and added this `info` during debugging, and decided not to remove it, so we're getting one for each (successful) call anyway. Would two be so much worse? ;-)

Reply:
Actually, there's already a `info` level message after parsing the input succeeds:

```
        logger.info(
            "QueryControllers POST for user {}, prefix {}, session {}: ({} - {})",
            user,
            prefix,
            session_token,
            start,
            end,
        )
```